tern/__main__.py

```diff
@@ -72,7 +72,7 @@ def create_top_dir(working_dir=None):
 def do_main(args):
     '''Execute according to subcommands'''
     # set bind mount location if working in a container
-    rootfs.set_mount_dir(args.bind_mount, args.working_dir)
+    rootfs.set_mount_dir(args.non_linux, args.bind_mount, args.working_dir)
```

Comment:
I think we can consolidate these options:
1. args.working_dir and args.bind_mount mean essentially the same thing here - make the working directory different from the default. So I think the `args.bind_mount` option should be deprecated. Either way, we are changing the CLI. I am not sure if this kind of change requires a major release change. Thoughts?
2. I envision that in the future, we will need to support running tern in an unprivileged container. In that sense, `non_linux` may not be a good name for this option. How about `--driver` and give options `fuse, overlay` and make `overlay` the default? That way we can add the `rsync` or whatever else option there may be to run tern in an unprivileged way.

Reply:
1) I would agree that deprecating the bind mount option and combining it into a `--driver` option should not constitute a major release. Especially because this CLI change is in the Dockerfile and I think most people build the Dockerfile provided in the repository, or copy the Dockerfile and slightly modify it, which won't affect what they have to interact with the CLI options.

2) I like the `--driver` suggestion. I will incorporate this.

Reply:
Combining the `bind_mount` and `non_linux` option will require the `--driver` option inputs to be a little funky. Folks will have to use either `--driver fuse` (which will use the default location inside the container) or `--driver /location/to/bind/mount`. @nishakm Does this sound ok?

Reply:
OK so here's my reasoning:
We have potentially 3 modes of operation and hence 3 ways a user can invoke tern:
| Natively on Linux  | In a privileged container | In an unprivileged container  |
|-------------------------|----------------------------------|---------------------------------------|
| --driver overlayfs  | --driver fuse                     | --driver rsync (or whatever)   |
| deps: none           | deps: fuse-overlayfs        | deps: whatever is required    |
| no external mnt    | external mnt if driver is overlayfs | no external mnt         |

In this situation I think avoiding the added complexity of checking to see if the driver choice is overlayfs and you are running in a container is a bit redundant because if you were running on a linux host, you could just tern natively. But if you were running in a container, it's less effort to just install fuse-overlay and not use a bind mount at all. This is why I think we should get rid of that option altogether.

Reply:
We could still allow the `-wd` option if someone wanted to point the working directory to a place other than the user's home directory.

Reply:
Ok, totally with you on getting rid of the `bind_mount` option. My question was more about having a --driver overlayfs option in addition to `--driver fuse`. If we are running in a container (regardless of Linux or non-linux host) and we just use the fuse-overlay to do the mounting, why does the user need to specify `--driver overlay` or `--driver fuse`? In either case, if we use fuse-overlayfs, we will be mounting inside the container.

Reply:
Oh I see what you're saying. We can default to the overlayfs mounting unless given the `--driver` option which can either be `fuse` or something else. I'm with you on this one. `overlayfs` need not be one of the options :+1:

Reply:
Cool! I will note in the "help" for the `--driver` CLI option that right now the only supported option for `--driver` is `fuse` but once we support non-privlidged containers we can change that note to expand for rsync, or whatever we want to call it.